storage/pebble/bootstrap.go

```diff
@@ -0,0 +1,136 @@
+package pebble
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/ledger/complete/wal"
+)
+
+type RegisterBootstrap struct {
+	checkpointDir      string
+	checkpointFileName string
+	log                zerolog.Logger
+	db                 *pebble.DB
+	leafNodeChan       chan *wal.LeafNode
+	rootHeight         uint64
+}
+
+// NewRegisterBootstrap creates the bootstrap object for reading checkpoint data and the height tracker in pebble
+// This object must be initialized and RegisterBootstrap.IndexCheckpointFile must be run to have the pebble db instance
+// in the correct state to initialize a Registers store.
+func NewRegisterBootstrap(
+	db *pebble.DB,
+	checkpointFile string,
+	rootHeight uint64,
+	log zerolog.Logger,
+) (*RegisterBootstrap, error) {
+	// check for pre-populated heights, fail if it is populated
+	// i.e. the IndexCheckpointFile function has already run for the db in this directory
+	checkpointDir, checkpointFileName := filepath.Split(checkpointFile)
```

Comment:
nit: this line should be moved bellow above the creation of struct, it's not related to checks